rpcs3/Emu/Cell/Modules/cellL10n.cpp

```diff
@@ -1275,10 +1306,86 @@ s32 UCS2stoEUCCNs()
 	return ConversionOK;
 }
 
-s32 SBCSstoUTF8s(vm::cptr<u8> src, vm::cptr<u32> src_len, vm::ptr<u8> dst, vm::ptr<u32> dst_len, s32 enc)
+s32 SBCSstoUTF8s(vm::cptr<u8> src, vm::ptr<u32> src_len, vm::ptr<u8> dst, vm::ptr<u32> dst_len, u32 code_page)
 {
-	cellL10n.warning("SBCSstoUTF8s(src=*0x%x, src_len=*0x%x, dst=*0x%x, dst_len=*0x%x, enc=*0x%x)", src, src_len, dst, dst_len, enc);
-	return _L10nConvertStr(enc, src, src_len, L10N_UTF8, dst, dst_len); // Might not work in some scenarios
+	cellL10n.notice("SBCSstoUTF8s(src=*0x%x, src_len=*0x%x, dst=*0x%x, dst_len=*0x%x, code_page=0x%x)", src, src_len, dst, dst_len, code_page);
+
+	HostCode code = 0;
+	if ((code_page >= _L10N_CODE_) || !_L10nCodeParse(code_page, code))
+	{
+		return ConverterUnknown;
+	}
+
+	ensure(src_len && dst_len); // Not really checked
+
+	if (*src_len == 0u)
+	{
+		*dst_len = 0;
+		return ConversionOK;
+	}
+
+	ensure(src); // Not really checked
+
+	u32 len = 0;
+	u32 dst_pos = 0;
+
+	for (u32 src_pos = 0; src_pos < *src_len; src_pos++)
+	{
+		u8 src_val = src[src_pos];
+		u64 longval = static_cast<u64>(src_val);
+		s32 utf8_len = 1;
+
+		if (static_cast<s8>(src_val) < '\0')
```

Comment:
Don't complain about my code when your code looks like this.

`if (static_cast<s8>(src_val) < '\0')`

If you want to implement these functions, at least make it readable.